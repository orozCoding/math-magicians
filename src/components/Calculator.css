#calc-container {
  width: 300px;
  text-align: center;
  justify-content: center;
  background-color: white;
}

#calc-results {
  width: 100%;
  color: white;
  background-color: rgb(115, 116, 141);
  text-align: end;
  padding: 15px;
}

.calc-row {
  width: 100%;
}

.btn-grey,
.btn-orange {
  width: 25%;
  padding: 15px;
  border: 1px solid rgb(211, 211, 211);
  cursor: pointer;
}

.btn-grey {
  background-color: rgb(240, 240, 240);
}

.btn-grey:hover {
  background-color: rgb(255, 255, 255);
}

.btn-orange {
  background-color: orange;
}

.btn-orange:hover {
  background-color: rgb(255, 222, 160);
}

#calc-btn-0 {
  width: 50%;
}
