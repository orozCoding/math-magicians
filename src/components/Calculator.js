import React, { Component } from 'react';
import './Calculator.css';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div id="calc-container" className="d-flex col">
        <div id="calc-results">0</div>
        <div id="calc-line-1" className="calc-row d-flex">
          <button type="button" id="calc-btn-ac" className="btn-grey">AC</button>
          <button type="button" id="calc-btn-plusminus" className="btn-grey">+/-</button>
          <button type="button" id="calc-btn-percent" className="btn-grey">%</button>
          <button type="button" id="calc-btn-divide" className="btn-orange">รท</button>
        </div>
        <div id="calc-line-2" className="calc-row d-flex">
          <button type="button" id="calc-btn-7" className="btn-grey">7</button>
          <button type="button" id="calc-btn-8" className="btn-grey">8</button>
          <button type="button" id="calc-btn-9" className="btn-grey">9</button>
          <button type="button" id="calc-btn-times" className="btn-orange">x</button>
        </div>
        <div id="calc-line-3" className="calc-row d-flex">
          <button type="button" id="calc-btn-4" className="btn-grey">4</button>
          <button type="button" id="calc-btn-5" className="btn-grey">5</button>
          <button type="button" id="calc-btn-6" className="btn-grey">6</button>
          <button type="button" id="calc-btn-minus" className="btn-orange">-</button>
        </div>
        <div id="calc-line-4" className="calc-row d-flex">
          <button type="button" id="calc-btn-1" className="btn-grey">1</button>
          <button type="button" id="calc-btn-2" className="btn-grey">2</button>
          <button type="button" id="calc-btn-3" className="btn-grey">3</button>
          <button type="button" id="calc-btn-plus" className="btn-orange">+</button>
        </div>
        <div id="calc-line-5" className="calc-row d-flex">
          <button type="button" id="calc-btn-0" className="btn-grey">0</button>
          <button type="button" id="calc-btn-dot" className="btn-grey">.</button>
          <button type="button" id="calc-btn-equal" className="btn-orange">=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
