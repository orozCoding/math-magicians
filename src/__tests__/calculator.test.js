import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Renders calculator', () => {
  it('renders', () => {
    const tree = renderer
      .create(
        <Calculator />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return the sum of two number form UI', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('='));
    const result = screen.getByText('21');
    expect(result.innerHTML).toBe('21');
  });
});
