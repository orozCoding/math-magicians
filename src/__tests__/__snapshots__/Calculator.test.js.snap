// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders calculator renders 1`] = `
<div
  className="d-flex col"
  id="calc-container"
>
  <div
    id="calc-results"
  >
    0
    
    
  </div>
  <div
    className="calc-row d-flex"
    id="calc-line-1"
  >
    <button
      className="btn-grey"
      id="calc-btn-ac"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn-grey"
      id="calc-btn-plusminus"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn-grey"
      id="calc-btn-percent"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn-orange"
      id="calc-btn-divide"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="calc-row d-flex"
    id="calc-line-2"
  >
    <button
      className="btn-grey"
      id="calc-btn-7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn-grey"
      id="calc-btn-8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn-grey"
      id="calc-btn-9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn-orange"
      id="calc-btn-times"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="calc-row d-flex"
    id="calc-line-3"
  >
    <button
      className="btn-grey"
      id="calc-btn-4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn-grey"
      id="calc-btn-5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn-grey"
      id="calc-btn-6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn-orange"
      id="calc-btn-minus"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="calc-row d-flex"
    id="calc-line-4"
  >
    <button
      className="btn-grey"
      id="calc-btn-1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn-grey"
      id="calc-btn-2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn-grey"
      id="calc-btn-3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn-orange"
      id="calc-btn-plus"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="calc-row d-flex"
    id="calc-line-5"
  >
    <button
      className="btn-grey"
      id="calc-btn-0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn-grey"
      id="calc-btn-dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn-orange"
      id="calc-btn-equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
